pkg/sql/sem/builtins/builtins.go

```diff
@@ -209,6 +209,7 @@ var StartCompactionJob func(
 	collectionURI, incrLoc []string,
 	fullBackupPath string,
 	encryptionOpts jobspb.BackupEncryptionOptions,
+	executionLocality roachpb.Locality,
```

Comment:
With locality-aware support coming later down the line, I'm thinking it might be better to replace `encryptionOpts` and `executionLocality` with a single `tree.BackupOptions` instead and have `StartCompactionJob` handle the processing. It'll keep the parameter list support and require less changes if we add more support down the line.